node/node_test.go

```diff
@@ -34,9 +33,10 @@ func TestRun(t *testing.T) {
 	cometListener, err := net.Listen("tcp", cometHTTPAddress)
 	require.NoError(t, err)
 	app := testapp.NewTest(t, chainID.String())
-	ethstatedb := rawdb.NewMemoryDatabase()
+	ethstatedb := testutils.NewEthStateDB(t)
 	defer func() {
-		require.NoError(t, ethstatedb.Close())
+		require.NoError(t, ethstatedb.DiskDB().Close())
+		require.NoError(t, ethstatedb.TrieDB().Close())
```

Comment:
Why do we need to close these manually? Is this documented anywhere?

Reply:
Looking at this again, I can see why this might be necessary. It still feels confusing that we would need to close them both ourselves (I would expect to use something like `ethstatedb.Close()`). 

Can you find any examples of this pattern in geth itself? I see cases these `Close` methods are not called.

Reply:
I'm not sure, the original code had this
```
ethstatedb := rawdb.NewMemoryDatabase()
	defer func() {
		require.NoError(t, ethstatedb.Close())
	}()
```
So I just followed the convention

Reply:
That's closing the rawdb, which we do in `testutils.NewEthStateDB`. I'm curious if we need to close the dbs returned on the `state.Database` interface.

Reply:
Here's some example from geth: https://github.com/ethereum/go-ethereum/blob/83e70aa3d00bbee9713b31768a30a7741eee8945/core/state/statedb_test.go#L277 and https://github.com/ethereum/go-ethereum/blob/83e70aa3d00bbee9713b31768a30a7741eee8945/eth/state_accessor.go#L71

The dbs returned by `state.Database` is not closed

Reply:
This one nerd-sniped me. It looks like the correct way to manage and close a `state.Database` is undocumented.

The `DiskDB()` is the underlying database, the `ethdb.Database` in our case. We defer the close when we open it ourselves.

It appears the TrieDB, on the other hand, _does_ need to be closed. Although there are a few places where geth does not close the TrieDB, they are all ephemeral (genesis state handling, for example). The trie db that get maintains on its `BlockChain` object is closed when the blockchain is stopped.

This means our current way of creating a `state.Database` error-prone (it's easy to forget to close the `TrieDB`). As a remedy, we can instead use the `state.NewDatabaseWithNodeDB` function to pass in the triedb separately.

I'll create a separate issue for this. For now @Jesserc, go ahead and remove the line that closes the `DiskDB`, but leave the line that closes the `TrieDB`. This seemingly innocuous change led to some important realizations! Thanks for your work here.

cc @natebeauregard this may be interesting and relevant for you in future withdrawals work.